### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by yoavamit**
Add basic unit testing to the `pivit` library.

## `fakeYubikey`
The `Pivit` interface defines the same public functions that are in the `piv.Yubikey` struct.
So both `pivYubikey` and `fakeYubikey` implement the `Pivit` intreface, but `fakeYubikey` does that using software only.

This implementation is nor secure in any way and should be used for testing only (which is why it's not exposed publicly).

## Missing tests

* `ImportCertificate` is also not tested, though that functionality is tested as a side effect in other tests
* There's also a bunch of edge cases, and smaller functions I didn't add tests for yet

### CODE REVIEW

File: pkg/pivit/sign_test.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+			}
+		})
+	}
+}
+
+// skipCI skips a test if we can determine the environment we're running in is a NixOS sandbox
+// it's used to skip tests that use networking for example
+func skipCI(t *testing.T) {
+	if os.Getenv("NIX_ENFORCE_PURITY") != "" {
+		t.Skip("Skipping test in CI environment")
```
Comment (linuxwolf) [inline pkg/pivit/sign_test.go:135]: curious, what about this test requires networking?

**Review by linuxwolf — APPROVED** [review]
one question; regardless of the answer this looks good to me!

File: pkg/pivit/sign_test.go (hunk 1 above)
Comment (yoavamit) [reply pkg/pivit/sign_test.go:135]: The timestamp URL - there's one test case that specifies a real timestamp authority server to include in the signature.
